genMarketplace.js

```diff
@@ -85,6 +85,91 @@ function createReadmeLink(listItem, itemType) {
   return Promise.resolve();
 }
 
+function travelDependenciesJson(firstLvlDepsJson, depsJson, startKey) {
+  /* Travels over the dependencies json to create a depdendencies map for each entry starting with startKey like so:
+  {
+    startKey: {
+      mandatory: {
+        packName: {
+          version: version
+        }
+      },
+      optional: {
+        packName: {
+          version: version
+        }
+      },
+      version: "...",
+      "mandatoryCount": ...,
+      "optionalCount": ...
+    }
+  }
+  */
+
+  if (startKey in depsJson === false) {
+    travelDependencies(depsJson, startKey, firstLvlDepsJson);
+    depsJson[startKey]['mandatoryCount'] = Object.keys(depsJson[startKey]['mandatory']).length;
+    depsJson[startKey]['optionalCount'] = Object.keys(depsJson[startKey]['optional']).length;
+  }
+}
+
+function travelDependencies(depsJson, startKey, firstLvlDepsJson) {
+  // Travel over the dependencies json of a given Pack dependency type, while collecting all sub-dependecy mandatory packs
+  
+  if (startKey in depsJson === false) {
+    depsJson[startKey] = {...firstLvlDepsJson[startKey]};
+  }
+
+  dependencyPacks = depsJson[startKey]['mandatory'];
+  for (var depKey in depsJson[startKey]['mandatory']) {
+    if (depsJson[depKey] === undefined) {
+      travelDependenciesJson(firstLvlDepsJson, depsJson, depKey);
+    }
+    
+    // fill mandatory sub-dependecies
+    for (var subDepKey in depsJson[depKey]['mandatory']) {
+      // skip if subDepKey is startKey or if subDepKey is already in node's collected mandatory
+      if (subDepKey !== startKey && (depsJson[startKey]['mandatory'][subDepKey] === undefined) ) {
+        if (depsJson[subDepKey] === undefined) {
+          // subDepKey wasn't yet traveled
+          travelDependenciesJson(firstLvlDepsJson, depsJson, subDepKey);
+        }
+        // collect subDepKey chained mandatory dependecies (subDepKey incl.)
+        for (var chainKey in getMandatoryChainDependencies(subDepKey, depsJson)) {
+          depsJson[startKey]['mandatory'][chainKey] = {
+            version: depsJson[chainKey].version,
+            support: depsJson[chainKey].support === "xsoar" ? "Cortex XSOAR" : capitalizeFirstLetter(depsJson[chainKey].support)
+          }
+        }
+      }
+    }
+  }
+}
+
+function getMandatoryChainDependencies(packName, depsJson, exclusionSet) {
+  // collects all mandatory chained dependecies, while skipping over keys it collected along the way
+
+  res = {};
+  if (packName in depsJson === false) {  // sanity
```

Comment:
What do you mean sanity? when will this happen?
```suggestion
  if !(packName in depsJson) {  // sanity
```

Reply:
it won't happen - this is why it's sanity, in case something unexpected happens.